### PR TITLE
DHP - Disconnect a device

### PR DESCRIPTION
**Pull request by reblg**
## Description
As a Veteran, I want to disconnect a connected device, so that I am not sending data from that device to the VA anymore. 

## Testing done
- Unit testing

## Screenshots
Screen Shot 2022-05-20 at 12 12 04 PM
Screen Shot 2022-05-20 at 12 11 53 PM

## Acceptance criteria

- [x] A Veteran has a connected device and has navigated to the Connected Devices page, the Veteran selects the Disconnect link underneath a connected device, the Disconnect Device modal is displayed with:the following message: Disconnecting your Fitbit will stop sharing data with VA the Disconnect device button the Go back button

- [x] A Veteran has selected the Disconnect link underneath a connected device, the Veteran selects the Go back button in the Disconnect Device modal, the Disconnect Device modal closes and the Connected Devices page is displayed

- [x] A Veteran has selected the Disconnect link underneath a connected device, the Veteran selects the Disconnect device button in the Disconnect Device modal, the Connected Devices page is displayed with:the Device Disconnected success message: Your device is no longer sharing data with VA the disconnected device no longer in the Your connected devices list the disconnected device listed in the Devices you can connect list

- [x] A Veteran selects the Disconnect button in the Disconnect modal, VetsAPI returns an HTTP error, The following error message is displayed

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by yonidoronpeters — APPROVED** [review]
💯  Looks great! just a couple of potential cleanup items

File: src/applications/dhp-connected-devices/components/ConnectedDevicesSection.jsx [hunk 1]
```diff
@@ -43,11 +34,42 @@ export const ConnectedDevicesSection = ({
...
+        />
+      )}
+      {disconnectFailureAlert && (
+        <DeviceConnectionAlert
+          testId="disconnection-failure-alert"
+          status="error"
+          headline="We couldn't disconnect your device"
+          description="We were not able to disconnect your device right now. Please try again."
+        />
+      )}
```
[the remark this answers is not in the document]
Comment (reblg) [reply src/applications/dhp-connected-devices/components/ConnectedDevicesSection.jsx:72]: I say tech debt card.

File: src/applications/dhp-connected-devices/components/DeviceDisconnectionCard.jsx [hunk 2]
```diff
@@ -1,30 +1,59 @@
...
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
+import environment from 'platform/utilities/environment';
+import { DisconnectModal } from './DisconnectModal';
+
+export function navigateToDisconnect(disconnectUrl) {
+  window.location.assign(
+    `${environment.API_URL}/dhp_connected_devices${disconnectUrl}`,
+  );
+}
```
[the remark this answers is not in the document]
Comment (reblg) [reply src/applications/dhp-connected-devices/components/DeviceDisconnectionCard.jsx:10]: No, and I'll remove it.

File: src/applications/dhp-connected-devices/components/DeviceDisconnectionCard.jsx [hunk 3]
```diff
@@ -1,30 +1,59 @@
...
+          {device.name}{' '}
+          <span className="connected-header-text"> - Connected</span>{' '}
+        </h3>
+        <p className="vads-u-margin-y--0">
+          <button
+            type="button"
+            onClick={() => setModalVisible(true)}
+            data-testid={`${device.key}-disconnect-link`}
+            id={`${device.key}-disconnect-link`}
+            onKeyDown={() => setModalVisible(true)}
```
[the remark this answers is not in the document]
Comment (reblg) [reply src/applications/dhp-connected-devices/components/DeviceDisconnectionCard.jsx:46]: Same thing, but one is for accessibility / keyboard users

File: src/applications/dhp-connected-devices/components/DeviceDisconnectionCard.jsx [hunk 4]
```diff
@@ -1,30 +1,59 @@
...
+
+export const DeviceDisconnectionCard = ({ device }) => {
+  const [modalVisible, setModalVisible] = useState(false);
+
+  function closeModal() {
+    setModalVisible(false);
+  }
+
+  function disconnectDevice() {
+    setModalVisible(false);
```
[the remark this answers is not in the document]
Comment (reblg) [reply src/applications/dhp-connected-devices/components/DeviceDisconnectionCard.jsx:20]: good catch will update

File: src/applications/dhp-connected-devices/components/DeviceDisconnectionCard.jsx [hunk 5]
```diff
@@ -1,30 +1,59 @@
...
+    window.location.href = `${environment.API_URL}/dhp_connected_devices${
+      device.disconnectUrl
+    }`;
```
[the remark this answers is not in the document]
Comment (reblg) [reply src/applications/dhp-connected-devices/components/DeviceDisconnectionCard.jsx:23]: Left over test code will delete 7-9

File: src/applications/dhp-connected-devices/components/DeviceDisconnectionCard.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (yonidoronpeters) [reply src/applications/dhp-connected-devices/components/DeviceDisconnectionCard.jsx:46]: oh cool, didn't know that!

File: src/applications/dhp-connected-devices/components/ConnectedDevicesSection.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (yonidoronpeters) [reply src/applications/dhp-connected-devices/components/ConnectedDevicesSection.jsx:72]: Created https://vajira.max.gov/browse/DHP-286

**Pull request merged**